### PR TITLE
Add a new target kakuteF4wing

### PR DESCRIPTION
**Pull request by jamming**
KakuteF4wing is made smaller and lighter. It cover an area of 25x30mm.
1) 6 PWM channel are: 2 motor pads and 4 servo with THT
2) 5 serial ports, one with inverter on the RX of a UART for traditional SBUS
3) 1 external I2C 
4) analog OSD
5) Independent 5V 3A BEC for servo
6)16MB onboard flash memory
7) Based on STM32F405
8) Simple PDB with 2BECs (5V/3A)
9) 2 ADC
10) Shunt resistance up to 90A.

### CODE REVIEW

File: configs/KAKUTEF4WING/config.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    TIMER_PIN_MAP( 3, PB1  , 2,  0) \
+    TIMER_PIN_MAP( 4, PC8  , 2,  0) \
+    TIMER_PIN_MAP( 5, PC9  , 2,  0) \
+    TIMER_PIN_MAP( 6, PA1  , 2,  0)
+
+
+#define ADC1_DMA_OPT        1
+
+#define MAG_I2C_INSTANCE (I2CDEV_1)
+#define USE_BARO
```
Comment (haslinghuis) [inline configs/KAKUTEF4WING/config.h:98]: Duplicate define.

```suggestion
```

**Review by haslinghuis — APPROVED** [review]

**Comment by ot0tot** [comment]
Which pins are motor outputs and which are servo outputs? Should the servos be assigned as a `SERVO` resource instead of a `MOTOR` resource? 

As-is, motors 1-4 conflict with DMA for SPI3 (flash), which will cause poor blackbox performance if they are used for DShot output. 

Also, LED_STRIP on PA1 has a DMA conflict with SPI2_TX, which will negatively impact analog OSD when LED_STRIP is enabled.

**Comment by jamming** [comment]
I am not aware that FLASH and OSD also need to use DMA. This board is currently a prototype board, and I will change the pinmap and test them in the next two days to ensure that TIMER and all SPIs have DMA. 
The new DMA MAP will be as follows:
# DMA1 Stream 0  :  SPI3_RX      :FLASH
# DMA1 Stream 2  :  TIM3_CH4  :PWM4
# DMA1 Stream 3  :  SPI2_RX      :OSD
# DMA1 Stream 4  :  SPI2_TX      :OSD
# DMA1 Stream 5  :  SPI3_TX      :FLASH
# DMA1 Stream 6  :  TIM2_CH2  :LED_STRIP
# DMA1 Stream 7  :  TIM3_CH3  :PWM3

# DMA2 Stream 0  :  SPI1_RX       :IMU
# DMA2 Stream 1  :  TIM1_CH1  :PWM1
# DMA2 Stream 2  :  TIM8_CH3  :PWM5
# DMA2 Stream 4  :  ADC1          :BAT_I/V
# DMA2 Stream 5  :  SPI1_TX      :IMU
# DMA2 Stream 6  :  TIM1_CH2  :PWM2
# DMA2 Stream 7  :  TIM8_CH4  :PWM6

Thank you for your help!

> Which pins are motor outputs and which are servo outputs? Should the servos be assigned as a `SERVO` resource instead of a `MOTOR` resource?
> 
> As-is, motors 1-4 conflict with DMA for SPI3 (flash), which will cause poor blackbox performance if they are used for DShot output.
> 
> Also, LED_STRIP on PA1 has a DMA conflict with SPI2_TX, which will negatively impact analog OSD when LED_STRIP is enabled.

**Comment by sugaarK** [comment]
https://betaflight.com/docs/development/manufacturer/config-target-guidance

please read and follow @jamming

**Comment by sugaarK** [comment]
sorry @jamming you guys are now part of the partner program so the new scheme other than working out resource and timer/DMA conflicts doesn't apply to you guys

**Review by KarateBrot — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**